Compilation fails on Windows-mingw64

I came from [this link](https://github.com/fortran-lang/fpm/pull/676#pullrequestreview-965457802), just out of curiosity, I used mingw64-gfortran to compile `tblite` on windows and found an unexpected error, I am not sure if it is a bug of mingw64-gfortran, but it can be done by replacing the `setenv` routine with `putenv` and solved ([see my code](https://github.com/zoziha/tblite/commit/16dc4a34d35cbf2124c1b2132174198300ec3d72)), the code becomes less, and it seems to be able to compile on windows-ifort and wsl2-gfortran (not very sure)?
```sh
>> fpm build
...
main.f90                               done.
tblite.exe                             failed.
[ 100%]Compiling...
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: build\gfortran_232594290C1E0524\tblite\libtblite.a(src_tblite_os.F90.o): in function `__tblite_os_MOD_setenv':C:\Users\zoziha\com.zoziha\fortran\tmp\tblite/./src/tblite/os.F90:69: undefined reference to `setenv'     
collect2.exe: error: ld returned 1 exit status
<ERROR> Compilation failed for object " tblite.exe "
<ERROR>stopping due to failed compilation
STOP 1
```

I'm not familiar with `xblite`, probably `xblite` is not intended to support windows-gfortran, if so, please ignore this issue.

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -70,6 +70,12 @@ jobs:
           compiler: intel
           version: 2021.2.0
 
+        - os: windows-latest
+          build: meson
+          build-type: debug
+          compiler: gnu
+          shell: msys2 {0}
+
     defaults:
       run:
         shell: ${{ matrix.shell || 'bash' }}
@@ -219,6 +225,7 @@ jobs:
         MESON_ARGS: >-
           ${{ matrix.compiler == 'intel' && '-Dfortran_link_args=-qopenmp' || '' }}
           ${{ contains(matrix.os, 'macos') && '-Dlapack=openblas' || '' }}
+          ${{ contains(matrix.os, 'windows') && '-Dfortran_link_args=-Wl,--allow-multiple-definition' || '' }}
 
     - name: Configure build (CMake)
       if: ${{ matrix.build == 'cmake' }}

--- a/src/tblite/os.F90
+++ b/src/tblite/os.F90
@@ -24,49 +24,25 @@ module tblite_os
 
 
    interface
-#ifndef _WIN32
-      function sys_setenv(name, value, overwrite) result(stat) &
-            & bind(c, name="setenv")
+      function sys_putenv(name) result(stat) &
+            & bind(c, name="putenv")
          import :: c_char, c_int
          character(len=c_char), intent(in) :: name(*)
-         character(len=c_char), intent(in) :: value(*)
-         integer(c_int), value :: overwrite
          integer(c_int) :: stat
       end function
-#else
-      function win_setenv(name, value) result(stat) &
-            & bind(c, name="SetEnvironmentVariable")
-         import :: c_char, c_int
-         character(len=c_char), intent(in) :: name(*)
-         character(len=c_char), intent(in) :: value(*)
-         integer(c_int) :: stat
-      end function
-#endif
    end interface
 
 
 contains
 
 
-#ifdef _WIN32
-function sys_setenv(name, value, overwrite) result(stat)
-   character(len=c_char), intent(in) :: name(*)
-   character(len=c_char), intent(in) :: value(*)
-   integer(c_int), value :: overwrite
-   integer(c_int) :: stat
-
-   stat = win_setenv(name, value)
-end function
-#endif
-
-
 subroutine setenv(name, value, stat)
    character(len=*), intent(in) :: name
    character(len=*), intent(in) :: value
    integer, intent(out) :: stat
-   integer(c_int), parameter :: overwrite = 1_c_int
 
-   stat = sys_setenv(as_c_char(name), as_c_char(value), overwrite)
+   stat = sys_putenv(as_c_char(name//"="//value))
+
 end subroutine setenv
 
 

